src/libraries/System.Console/src/System/IO/SyncTextReader.cs

```diff
@@ -95,11 +96,21 @@ public override string ReadToEnd()
             return Task.FromResult(ReadLine());
         }
 
+        public override ValueTask<string?> ReadLineAsync(CancellationToken cancellationToken)
+        {
+            return cancellationToken.IsCancellationRequested ? ValueTask.FromCanceled<string?>(cancellationToken) : new ValueTask<string?>(ReadLine());
```

Comment:
Nit: formatting to make it a bit easier to read:
```suggestion
            return cancellationToken.IsCancellationRequested ?
                ValueTask.FromCanceled<string?>(cancellationToken) :
                new ValueTask<string?>(ReadLine());
```